packages/universal-components/src/RangeDatePicker/RenderDay.js

```diff
@@ -50,89 +62,150 @@ const RenderArrow = (props: RenderArrowProps) =>
+        onLayout={this.measureDayItem}
       >
-        <>
-          {isStartOfSelectedDates && isRangePicker && (
-            <RenderArrow
-              onPress={onArrowPress(onLeftPress)}
-              style={styles.leftArrow}
-              direction="left"
-            />
-          )}
-          <View
-            style={[
-              styles.dayContainer,
-              isDaySelected && styles.selectedDatesContainer,
-              isStartOfSelectedDates && styles.startOfSelectedDatesContainer,
-              isEndOfSelectedDates && styles.endOfSelectedDatesContainer,
-            ]}
-          >
-            {!isFieldEmpty && (
-              <>
-                <Text
-                  weight="bold"
-                  style={[
-                    styles.day,
-                    isDaySelected && styles.selectedDatesText,
-                    isDayInPast(day) && styles.dayInPast,
-                  ]}
-                >
-                  {format(day, 'd')}
-                </Text>
-                {price ?? <DayPrice price={price} />}
-              </>
+        <View>
+          {isRangePicker &&
+            (isStartOfSelectedDates || this.state.isDragging) && (
+              <DraggableItem
+                onDrag={this.onDrag}
+                onDrop={this.onDrop}
+                style={styles.draggableItemLeft}
+                grabbedSide="left"
+              />
             )}
-          </View>
-          {isEndOfSelectedDates && isRangePicker && (
-            <RenderArrow
-              onPress={onArrowPress(onRightPress)}
-              style={styles.rightArrow}
-              direction="right"
+
+          <Touchable
+            onPress={this.handlePress}
+            disabled={isDayInPast(day) || isFieldEmpty}
+          >
+            <>
+              {isStartOfSelectedDates && isRangePicker && (
+                <RenderArrow
+                  onPress={onArrowPress(this.onLeftPress)}
+                  style={styles.leftArrow}
+                  direction="left"
+                />
+              )}
+              <View
+                style={[
+                  styles.dayContainer,
+                  isDaySelected && styles.selectedDatesContainer,
+                  isStartOfSelectedDates &&
+                    styles.startOfSelectedDatesContainer,
+                  isEndOfSelectedDates && styles.endOfSelectedDatesContainer,
+                ]}
+              >
+                {!isFieldEmpty && (
+                  <>
+                    <Text
+                      weight="bold"
+                      style={[
+                        styles.day,
+                        isDaySelected && styles.selectedDatesText,
+                        isDayInPast(day) && styles.dayInPast,
+                      ]}
+                    >
+                      {format(day, 'd')}
+                    </Text>
+                    {price ?? <DayPrice price={price} />}
```

Comment:
price is of type `?string` if price is equal to empty string, this crashes on android.

Reply:
Thanks, I always forget about this one. Done

Reply:
Once you have spent 4 hours trying to find out why it works on ios and not on android, you never forget again 😄 🐘